src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public sealed class NuGetProjectManagerService : INuGetProjectManagerService
+    {
+        private readonly ServiceActivationOptions _options;
+        private readonly IServiceBroker _serviceBroker;
+        private readonly AuthorizationServiceClient _authorizationServiceClient;
+
+        public NuGetProjectManagerService(ServiceActivationOptions options, IServiceBroker sb, AuthorizationServiceClient ac)
+        {
+            _options = options;
+            _serviceBroker = sb;
+            _authorizationServiceClient = ac;
+        }
+
+        public async ValueTask<IReadOnlyCollection<string>> GetProjectsAsync(CancellationToken ct)
```

Comment:
Not sure how useful it is to a caller just to get project GUID's.

How about `GetProjectsAsync(...)` returning `IReadOnlyCollection<ProjectContextInfo>`?

Reply:
I was thinking about this, that would mean we would need to move ProjectContextInfo to internal.contracts or create an interface for it, thoughts?